### PR TITLE
feat(firestore): Partition samples

### PR DESCRIPTION
**Pull request by AlisskaPie**

### CODE REVIEW

File: firestore/firestore_snippets/partition_query.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+func partitionQuery(ctx context.Context, w io.Writer, parent string) error {
+	// projectID := "project-id"
+	client, err := firestore.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("firestore.NewClient: %v", err)
+	}
+	defer client.Close()
+
+	structuredQuery := &firestorepb.StructuredQuery{
+		Select: &firestorepb.StructuredQuery_Projection{}}
```
Comment (IlyaFaer) [inline firestore/firestore_snippets/partition_query.go]: The last `}` should be on the next line

**Comment by AlisskaPie** [comment]
Related python [PR](https://github.com/googleapis/python-firestore/pull/209/files#diff-a387a6bac6c791d19451400f5319c397R293),  java [PR](https://github.com/googleapis/java-firestore/pull/202/files).

Not sure about `collectionGroupID` record.
How do you think will be better?
1. `collectionGroupID := collectionID + "-subcollection"`
2. `collectionGroupID := tc.ProjectID + "-subcollection"`?

And `collectionGroupID` or maybe rename to `subcollection`?

Also, I'm not confident about this move: 
`parent := strings.Replace(collection.Path, "/"+collectionID, "", 1)`
Maybe this way will be easier to read:
`parent := fmt.Sprintf("projects/%v/databases/(default)/documents", projectID)`?

**Comment by IlyaFaer** [comment]
@AlisskaPie,

> How do you think will be better?
> 
> collectionGroupID := collectionID + "-subcollection"
> collectionGroupID := tc.ProjectID + "-subcollection"?

`collectionID` already include `tc.ProjectID`, so the first variant looks better.

> Maybe this way will be easier to read:
> parent := fmt.Sprintf("projects/%v/databases/(default)/documents", projectID)?

Yes

File: firestore/firestore_snippets/partition_query.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+// [START fs_partition_query]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	firestore "cloud.google.com/go/firestore/apiv1"
+	"google.golang.org/api/iterator"
+	pb "google.golang.org/genproto/googleapis/firestore/v1"
```
Comment (IlyaFaer) [inline firestore/firestore_snippets/partition_query.go:25]: Aren't these imports sorted?

File: firestore/firestore_snippets/partition_query_test.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+	"testing"
+
+	"cloud.google.com/go/firestore"
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+)
+
+func TestPartion(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	ctx := context.Background()
+	projectID := os.Getenv("GOLANG_SAMPLES_FIRESTORE_PROJECT")
```
Comment (IlyaFaer) [inline firestore/firestore_snippets/partition_query_test.go:32]: I think, we could use just `tc.ProjectID` here

File: firestore/firestore_snippets/partition_query.go (hunk 2 above)
Comment (AlisskaPie) [reply firestore/firestore_snippets/partition_query.go:25]: Yes, autosorted

File: firestore/firestore_snippets/partition_query_test.go (hunk 3 above)
Comment (AlisskaPie) [reply firestore/firestore_snippets/partition_query_test.go:32]: I guess we should keep it as it is, see [this comment](https://github.com/GoogleCloudPlatform/golang-samples/pull/1618#discussion_r480078594)

**Review by AlisskaPie — COMMENTED** [review]
Thanks for review

**Pull request closed without merging**